### PR TITLE
[WIP] Add ring buffer settings

### PR DESCRIPTION
**Pull request by improbable-valentyn**
#### Description
This is a small part of ring buffer work that doesn't do much on its own but can be reviewed separately.

#### Primary reviewers
@m-samiec @aleximprobable

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Public/SpatialGDKSettings.h [hunk 1]
```diff
@@ -207,4 +207,20 @@ class SPATIALGDK_API USpatialGDKSettings : public UObject
...
+
+	UPROPERTY(EditAnywhere, Config, Category = "Replication", meta = (DisplayName = "Use RPC Ring Buffers"))
+	bool bUseRPCRingBuffers;
+
+	UPROPERTY(EditAnywhere, Config, Category = "Replication", meta = (EditCondition = "bUseRPCRingBuffers", DisplayName = "Default RPC Ring Buffer Size"))
+	uint32 DefaultRPCRingBufferSize;
+
+	/** Overrides default ring buffer size. */
+	UPROPERTY(EditAnywhere, Config, Category = "Replication", meta = (EditCondition = "bUseRPCRingBuffers", DisplayName = "RPC Ring Buffer Size Map"))
+	TMap<ERPCType, uint32> RPCRingBufferSizeMap;
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDK/Public/SpatialGDKSettings.h]: Not sure how to limit the key to client/server reliable/unreliable and multicast (currently also allows setting the key to "Invalid" and "Cross Server", which looks odd but doesn't really affect anything).

File: SpatialGDK/Source/SpatialGDK/Public/SpatialGDKSettings.h [hunk 2]
```diff
@@ -207,4 +207,22 @@ class SPATIALGDK_API USpatialGDKSettings : public UObject
 
 	UPROPERTY(EditAnywhere, config, Category = "Load Balancing", meta = (EditCondition = "bEnableUnrealLoadBalancer"))
 	TSubclassOf<class UAbstractLBStrategy> LoadBalanceStrategy;
+
+	UPROPERTY(EditAnywhere, Config, Category = "Replication", meta = (DisplayName = "Use RPC Ring Buffers"))
+	bool bUseRPCRingBuffers;
+
+private:
+	UPROPERTY(EditAnywhere, Config, Category = "Replication", meta = (EditCondition = "bUseRPCRingBuffers", DisplayName = "Default RPC Ring Buffer Size"))
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Public/SpatialGDKSettings.h]: nit: these display names are redundant

Comment (improbable-valentyn) [reply SpatialGDK/Source/SpatialGDK/Public/SpatialGDKSettings.h]: Not quite. This is what it looks like without specifying the display name:
image

Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Public/SpatialGDKSettings.h]: The horror

**Pull request merged**